tests/test_krakenz3_response.py

```diff
@@ -1,4 +1,38 @@
 # fmt:off
+from PIL import Image, ImageSequence
+import os
+import io
+""" 
+Required for different library versions
+"""
+def load_rgb_gif():
```

Comment:
This is doing the same thing as the driver, right? And then verifying if it's the same - IMO this defeats the purpose of the test. It should contain the hardcoded values, so we're sure that those are generated and sent. (But that got disrupted by pillow changing the way they generate the output.)

@jonasmalacofilho, thoughts?

Reply:
There is another way of thinking about it, everything is static except the data that library is producing, and we are not testing library.
Granted, the way that we are using the library might be wrong, but it is proven to work that way.

Reply:
I can revert this if you like.
But even if we generate gif file, there is no guarantee that gif will be same between versions.

Reply:
Actually, what @sophipl just wrote gave me an idea...

During the test, let's inject an already appropriately massaged image into the driver at the right place. This way, when testing the driver, pillow (or whatever we choose to use as a image manipulation library) doesn't matter.

Ideally we also want to check that the driver _tries_ to resize the image, so maybe do something like:

- the driver calls a "normalize image" function, which normally uses pillow, and that can be easily tested in isolation (supply it with a pillow-generated GIF of the wrong size, and have its output match the pillow-generated equivalent GIF of the right size);
- when testing the rest of the driver, specifically the `set_screen` functionality, the "normalize image" is monkeypatched to just return an already well formed GIF that the driver can just split and send to the (mock) device.

This would mean retaining a binary blog in the repository, which (as mentioned before) is not ideal. So let's keep this binary blob as simple as possible: like, 3 frames, each of a solid base color. Something that's really simple to audit; it doesn't have to look good, it just has to produce a corresponding golden sample of traffic that's useful enough for us to have some confidence in the driver.

Reply:
I will revert this commit then.
It seems that bigger test rewrite is in order with mocking the library if I understand the idea.
Unfortunately new firmware does not support gifs, as far as I know, so the test is related to the functionality added.